components/DeleteNodeDialog.tsx

```diff
@@ -48,35 +47,48 @@ export function DeleteNodeDialog(props: {
   if (!props.visible) return null;
 
   const handleClose = () => props.onClose();
-  const handleDelete = () =>
+  const handleDelete = (includeChildren: boolean) =>
     deleteMutation.mutate({
       id: props.objectToDelete.id,
       projectId: props.objectToDelete.projectId,
-      type: props.objectToDelete.type,
+      includeChildren: includeChildren,
     });
 
-  const { type } = props.objectToDelete;
+  const { type, children } = props.objectToDelete;
+  const hasChildren = children.length > 0;
 
   const header = `Delete ${getNodeTypeName(type).toLowerCase()}`;
-  let warningMessage = "This will delete the selected card.";
-  if (type === mainActivity) {
-    warningMessage =
-      "This will delete everything under it.\nAre you sure you want to proceed?";
-  } else if (type === choice) {
-    warningMessage =
-      "This will delete all connected alternatives.\nAre you sure you want to proceed?";
-  }
+
+  const getWarningMessage = () => {
+    if (type === mainActivity && hasChildren) {
+      return "This will delete all of its following cards.\nAre you sure you want to proceed?";
+    } else if (type === choice && hasChildren) {
+      return "This will delete all connected alternatives.\nAre you sure you want to proceed?";
+    }
+    return "This will delete the selected card";
+  };
+
+  const getCheckboxMessage = () => {
+    if ((type === subActivity || type === waiting) && hasChildren) {
+      return "Delete all of its following cards";
+    }
+  };
```

Comment:
I would move this to a separate file to reduce the length of this component.

Reply:
Since these functions will not be used anywhere else and they are pretty small I think it's fine keeping them here. It's faster to read the component's code without having to go to other files and the component is not very big either.